app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ApplicationRecord
+  before_save {email.downcase!}
```

Comment:
Method `downcase!` vs `downcase` có gì khác nhau?

Reply:
`downcase!` trả về chuỗi toàn chữ thường và thay đổi chính giá trị của chuỗi trong object, còn `downcase` chỉ trả về chuỗi toàn chữ thường thôi

Reply:
còn gì nữa không?

Reply:
`downcase!` trả về `nil` nếu không có thay đổi, `downcase` thì chỉ trả về chuỗi thôi kể cả không có thay đổi